lib/form-persister.js

```diff
@@ -0,0 +1,33 @@
+export default class FormPersister {
+  constructor(formId = null, throttlePeriod = 1000) {
+    this.formId = formId;
+    this.throttlePeriod = throttlePeriod;
+    this.canSave = true;
+  }
+
+  setFormId(formId) {
+    this.formId = formId;
+  }
+
+  saveValues(values) {
+    if (this.canSave && this.formId) {
+      this.canSave = false;
+      window.localStorage.setItem(this.formId, JSON.stringify(values));
+      setTimeout(() => (this.canSave = true), this.throttlePeriod);
```

Comment:
Throttling is not easy, but in this case we're losing the last 1 seconds of typing. Could that be problematic?

Reply:
I could tweak this to update the localStorage when the timer completes